pkg/deployment/deployment.go

```diff
@@ -84,10 +84,20 @@ func (d *Deployer) Deploy(services []string) (*ctrl.Result, error) {
 		d.AeeSpec.ExtraMounts = make([]storage.VolMounts, len(aeeSpecMounts))
 		copy(d.AeeSpec.ExtraMounts, aeeSpecMounts)
 		d.AeeSpec, err = d.addServiceExtraMounts(foundService)
-
 		if err != nil {
 			return &ctrl.Result{}, err
 		}
+
+		// Add certMounts to the install-certs service if TLS enabled
+		if *d.NodeSet.Spec.TLSEnabled {
+			if foundService.Name == InstallCertsLabel {
```

Comment:
This is more service specific logic in the code, which I really don't think is needed. I left this comment on the initial PR, but agreed to allow it to move forward for progress sake, ... but nova/libvirt is still hardcoded. Now this hardcodes something for install-certs. We need to find a different way.

Off the top of my head, an AddCertMounts bool on the Service type, could be a bool used to trigger this logic. This would also keep this logic functional when using non-default services. We don't want restrictions like "you must have a service called install-certs if you have any other services that use certs". Instead, the bool provides a more usable interface to enable/disable this functionality.

Otherwise, this looks ok to me as a general direction. Although, it would be nice if we could eliminate the extra loop of fetching all the services somehow.

Reply:
ACK.  Your initial comments about the service specific code hasn't been forgotten.  The issue has been captured in https://issues.redhat.com/browse/OSPRH-2389 which is on-deck to be tackled shortly.

Will add a boolean for this, as suggested.

Reply:
See https://github.com/openstack-k8s-operators/dataplane-operator/pull/610 for the changes to resolve your concerns with cert contents